test/csuite/random_directio/main.c

```diff
@@ -887,6 +888,21 @@ kill_child(pid_t pid)
     testutil_assert_errno(waitpid(pid, &status, 0) != -1);
 }
 
+/*
+ * die --
+ *     Called when testutil_assert or testutil_check fails to clean up a child process if it exists.
+ */
+static void
+die(void)
+{
+    pid_t pid;
+
+    pid = child_pid;
+    child_pid = 0;
+    if (pid != 0)
+        kill_child(pid);
```

Comment:
Are you confident that in every scenario this can be invoked if a child process exists it is suspended?
As `kill_child()` makes this assumption.

Reply:
I've read through the code and believe it will be in a suspended state.
But doesn't the `child_pid` need to be reset to 0 at approx L1394 where it has been killed?

Reply:
Yes it should be reset.